docs/guide/custom-rules.md

```diff
@@ -131,6 +131,34 @@ These rules require at least one argument and the target field must have a match
 <input name="passwordConfirmation" ref="confirmation" type="password" placeholder="Confirm the password">
 ```
 
+## Require-like rules
+
+If you need to define a rule that declares a field as required under specific conditions, you will need to enable the `computesRequired` option. Because, by default, VeeValidate doesn't validate non require-like rules if the field has no value. The `computesRequired` will tell VeeValidate to run your rule even if the field has no value.
+
+In consequence, your rule **must** return an object with both `valid` and `data.required` boolean properties:
+```js
+validator.extend('my_custom_rule', (value, [otherValue]) => {
+  // do something and finally return an object like this one:
+  return {
+    valid: true, // or false
+    data: {
+      required: true // or false
+    }
+  };
+}, {
+  computesRequired: true
+});
+```
+
+Your rule also has to check for the requirement to be fulfilled.
+If `data.required` is set to false, the other field rules won't be executed if the field is empty.
+::: tip
+This option is best used when combined with `hasTarget`, to make a conditionnal requirement, based on other field values. Like the `require_if` rule.
+:::
+::: danger
+You cannot use this option if you want to write an async (promise-based) rule. You rule implementation has to be synchronous.
```

Comment:
Is there a reason for this limitation? I gather it's because of the other field value integrity during very long running validations.

Reply:
It's because I didn't want to alter / complexify the _validate function by adding a Promise.all(...).then procedure, which would probably add a level of indentation.

I'm very familiar with async/await stuff, very handy to use in those cases ; but I guessed that you didn't want to use them in this project(?), maybe because of the big polyfill that would represent in the built files(?)

Reply:
At some point I had `async/await` and had them transpiled using [nodent compiler](https://github.com/MatAtBread/nodent)

I reverted back because since the `async` method gets transpiled to a function that returns a promise, the `buble` would then transpile them to ES5 functions that do not correctly resolve the `this` context inside those promises. Might revisit the issue soon though.

can I merge the PR?

Reply:
Ah, ok, I see.

The PR is good to merge, we'll maybe revisit the async limitation later.